lib/lightstep/span_context.rb

```diff
@@ -1,12 +1,31 @@
+#frozen_string_literal: true
+
 module LightStep
   # SpanContext holds the data for a span that gets inherited to child spans
   class SpanContext
-    attr_reader :id, :trace_id, :baggage
+    attr_reader :id, :trace_id, :trace_id16, :sampled, :baggage
+    alias_method :sampled?, :sampled
+
+    ZERO_PADDING = '0' * 16
 
-    def initialize(id:, trace_id:, baggage: {})
+    def initialize(id:, trace_id:, sampled: true, baggage: {})
       @id = id.freeze
-      @trace_id = trace_id.freeze
+      @trace_id16 = pad_id(trace_id).freeze
+      @trace_id = truncate_id(trace_id).freeze
+      @sampled = sampled
       @baggage = baggage.freeze
     end
+
+    private
+
+    def truncate_id(id)
+      return id unless id && id.size == 32
+      id[16..-1]
+    end
+
+    def pad_id(id)
+      return id unless id && id.size == 16
+      "#{ZERO_PADDING}#{id}"
```

Comment:
I'm following the convention set out in the [JS PR](https://github.com/lightstep/lightstep-tracer-javascript/pull/177/files), but I am wondering if we should be padding on the right side of an 8 byte id. There was some [discussion](https://github.com/w3c/trace-context/issues/337) and a [recent PR](https://github.com/w3c/trace-context/pull/344) around this in the W3C Trace Context repo.

Reply:
I think, conventionally, we only pad the LHS of the id. I'm not sure what'd happen if you added the padding to the RHS - presumably it'd all convert to the same `uint64` or whatever.